### PR TITLE
fix(site): fix metadata sorting

### LINKED ISSUE #12348: Workspace agents not getting sorted by the order parameter but by display name instead

### PR DESCRIPTION
**Pull request by BrunoQuaresma**
Close https://github.com/coder/coder/issues/12348

### CODE REVIEW

File: coderd/workspaceagents.go [hunk 1]
```diff
@@ -1601,6 +1601,7 @@ func convertWorkspaceAgentMetadata(db []database.WorkspaceAgentMetadatum) []code
 	result := make([]codersdk.WorkspaceAgentMetadata, len(db))
 	for i, datum := range db {
 		result[i] = codersdk.WorkspaceAgentMetadata{
+			Order: datum.DisplayOrder,
```
Comment (BrunoQuaresma) [inline coderd/workspaceagents.go:1604]: How do I know if this value is always returned from `datum.DisplayOrder`? Does Go return `0` if the user didn't define it in the terraform config?

**Review by mtojek — CHANGES_REQUESTED** [review]
It should be fixed in this PR: https://github.com/coder/coder/pull/12066

You don't need to pass `order` up, as items returned by the endpoint should be in order. What is the issue you're trying to solve?

**Comment by BrunoQuaresma** [comment]
Looks like it is not sorted.

File: site/src/modules/resources/AgentMetadata.tsx [hunk 2]
```diff
@@ -34,9 +34,11 @@ export const AgentMetadataView: FC<AgentMetadataViewProps> = ({ metadata }) => {
   }
   return (
     <section css={styles.root}>
-      {metadata.map((m) => (
-        <MetadataItem key={m.description.key} item={m} />
-      ))}
+      {[...metadata]
+        .sort((a, b) => a.order - b.order)
```
Comment (mtojek) [inline site/src/modules/resources/AgentMetadata.tsx:38]: We sort these items on the backend side. We don't need to sort it one more time here. If there is a bug, it should be fixed on the backend side.

**Comment by BrunoQuaresma** [comment]
Close in favor of fixing this only on BE

**Pull request closed without merging**